code/modules/reagents/chemistry/reagents/alcohol.dm

```diff
@@ -1518,3 +1518,240 @@
+	if(isdrask(M))
+		if(current_cycle % 10 == 0 && prob(50))
+			to_chat(M, "<span class='warning'>Your skin emits a soapy liquid from its pores cleaning you in the process.</span>")
+			M.clean_blood()
+	return ..()
+
+/datum/reagent/consumable/ethanol/acid_dreams
+	name = "Acid Dreams"
+	id = "aciddreams"
+	description = "This one looks just wierd and reeks of acid."
+	color = "#B7FF6A" // rgb: 183, 255, 106
+	alcohol_perc = 0.7
+	drink_icon = "aciddreams"
+	drink_name = "Acid Dreams"
+	drink_desc = "This one looks just wierd and reeks of acid."
+	taste_description = "acid"
+
+/datum/reagent/consumable/ethanol/acid_dreams/on_mob_life(mob/living/M)
+	if(isgrey(M))
+		if(current_cycle % 10 == 0 && prob(50))
+			var/list/mob/living/targets = list()
+			for(var/mob/living/L in orange(14, M))
+				if(L.is_dead()) //we don't care about dead mobs
+					continue
+				if(!L.client)
+					continue
+				targets += L
+			var/mob/living/target = pick(targets)
+			if(target == null)
+				return ..()
+			to_chat(M, "<span class='warning'>You feel that [target.name] is somewhere near.</span>")
+
+	return ..()
+
+/datum/reagent/consumable/ethanol/islay_whiskey
+	name = "Islay Whiskey"
+	id = "islaywhiskey"
+	description = "Named in honor of one of the most gritty and earth smelling types of Whiskey of Earth, this drink is a treat for any Diona."
+	color = "#461300" // rgb: 70, 19, 0
+	alcohol_perc = 0.2
+	drink_icon = "islaywhiskey"
+	drink_name = "Islay Whiskey"
+	drink_desc = "Named in honor of one of the most gritty and earth smelling types of Whiskey of Earth, this drink is a treat for any Diona."
+	taste_description = "soil"
+
+/datum/reagent/consumable/ethanol/islay_whiskey/on_mob_life(mob/living/M)
+	if(isdiona(M))
+		if(current_cycle > 5)
+			var/mob/living/carbon/human/H = M
+			var/turf/T = H.loc
+			var/light_amount = min(1, T.get_lumcount()) - 0.5
+
+			if(light_amount > 0.2 && !H.suiciding)
+				if(H.health > 0)
+					H.adjustBruteLoss(-0.25)
+					H.adjustToxLoss(-0.25)
+					H.adjustOxyLoss(-0.25)
+	return ..()
+
+/datum/reagent/consumable/ethanol/ultramatter
+	name = "Ultramatter"
+	id = "ultramatter"
+	description = "In the triangle of fire, this is apex of fuel."
+	color = "#38004B" // rgb: 56, 0, 75
+	alcohol_perc = 0.7
+	drink_icon = "ultramatter"
+	drink_name = "Ultramatter"
+	drink_desc = "In the triangle of fire, this is apex of fuel."
+	taste_description = "fire"
+	var/on_fire = FALSE
+
+/datum/reagent/consumable/ethanol/ultramatter/on_mob_life(mob/living/M)
+	if(isplasmaman(M))
+		if(on_fire)
+			M.adjust_fire_stacks(-1)
+			on_fire = FALSE
+		if(current_cycle % 10 == 0 && prob(30))
+			var/mob/living/carbon/human/H = M
+			to_chat(M, "<span class='warning'>You expell flaming substance from within your suit.</span>")
+			var/obj/item/clothing/under/plasmaman/suit = H.w_uniform
+			if(suit != null)
+				suit.next_extinguish = world.time + 10 SECONDS
+			H.adjust_fire_stacks(1)
+			H.IgniteMob()
+			on_fire = TRUE
```

Comment:
```suggestion
```
this will be handled automatically by `H.IgniteMob()`

Reply:
it's not Mobs "on_fire" it's the reagent's on_fire to know that next tick we should extinguish them

Reply:
op you're right